trino-aws-proxy-glue/src/main/java/io/trino/aws/proxy/glue/rest/ModelLoader.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.glue.rest;
+
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.reflect.ClassPath;
+import com.google.inject.Binder;
+import com.google.inject.TypeLiteral;
+import com.google.inject.multibindings.MapBinder;
+import software.amazon.awssdk.core.SdkPojo;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class ModelLoader
+{
+    private static final String REQUEST_SUFFIX = "Request";
+    private static final String REQUEST_PREFIX = "AWSGlue.";
+
+    private static final LoadedClasses loadedClasses = new LoadedClasses();
+
+    private static class LoadedClasses
+    {
+        private final Map<String, Class<?>> requestClasses;
+        private final Set<Class<?>> modelClasses;
+
+        private LoadedClasses()
+        {
+            Map<String, Class<?>> requestClasses = new HashMap<>();
+            Set<Class<?>> modelClasses = new HashSet<>();
```

Comment:
nit: `ImmutableSetBuilder`?

Reply:
I didn't want to deal with its limitations. We don't care, for example, if we get duplicate entries.

Reply:
Make sense